suitcase/spec.py

```diff
@@ -375,53 +435,104 @@ def spec_to_document(specfile, scan_ids=None, validate=False):
     If multiple scan_ids are requested, then documents will be emitted in the
     order listed above until all scan_ids have been processed
     """
-    if isinstance(specfile, str):
-        specfile = Specfile(filename=specfile)
-    if isinstance(specfile, Specfile):
-        # grab the scans that we want to convert
-        if scan_ids is None:
-            scans_to_process = specfile
-        else:
-            try:
-                scan_ids[0]
-            except TypeError:
-                scan_ids = [scan_ids]
-            finally:
-                scans_to_process = [specfile[sid] for sid in scan_ids]
-    elif isinstance(specfile, Specscan):
-        scans_to_process = [specfile]
-    else:
-        raise ValueError("Variable `specfile` is of type {0}. We only support"
-                         "strings, suitcase.spec.Specfile or "
-                         "suitcase.spec.Specscan objects here."
-                         "".format(specfile))
+    raise ValueError("Variable `specfile` is of type {0}. We only support"
+                     "strings, suitcase.spec.Specfile or "
+                     "suitcase.spec.Specscan objects here."
+                     "".format(specfile))
+
 
+@spec_to_document.register(str)
+def _(specfile, scan_ids=None, validate=False, check_in_broker=False):
+    """
+    Handle the case when type(specfile) == str
+    """
+    specfile = Specfile(filename=specfile)
+    for document_name, document in spec_to_document(
+            specfile, scan_ids=scan_ids, validate=validate,
+            check_in_broker=check_in_broker):
+      yield document_name, document
+
+
+@spec_to_document.register(Specfile)
+def _(specfile, scan_ids=None, validate=False, check_in_broker=False):
+    """
+    Handle the case when type(specfile) == Specfile
+    """
+    # grab the scans that we want to convert
+    if scan_ids is None:
+        scans_to_process = specfile
+    else:
+        try:
+            scan_ids[0]
+        except TypeError:
+            scan_ids = [scan_ids]
+        finally:
+            scans_to_process = [specfile[sid] for sid in scan_ids]
     for scan in scans_to_process:
-        for document_name, document in specscan_to_document_stream(
-                scan, validate=validate):
-            yield document_name, document
+        try:
+            for document_name, document in spec_to_document(
+                    scan, scan_ids=scan_ids, validate=validate,
+                    check_in_broker=check_in_broker):
+                yield document_name, document
+        except NotImplementedError as e:
+            logger.debug(e.args[0])
```

Comment:
I'm not entirely sure. I defaulted to debug because there's nothing that the end user can do to fix this NotImplementedError.  I don't want this to raise the NotImplementedError here because that will kill the insertion of a specfile if this code encounters a `timescan` or any other spec scan types that I have not mapped yet.  

Any of the log levels are fine with me. What would you prefer?